packages/ai-chat/src/browser/session-summary-variable-contribution.ts

```diff
@@ -0,0 +1,93 @@
+// *****************************************************************************
+// Copyright (C) 2025 EclipseSource GmbH and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { AIVariable, AIVariableContext, AIVariableResolutionRequest, AIVariableResolver, ResolvedAIContextVariable } from '@theia/ai-core';
+import { AIVariableCompletionContext, FrontendVariableContribution, FrontendVariableService } from '@theia/ai-core/lib/browser';
+import { MaybePromise, QuickInputService, QuickPickItem, QuickPickItemOrSeparator, QuickPickSeparator } from '@theia/core';
+import { ChatService } from '../common';
+import { codiconArray } from '@theia/core/lib/browser';
+import * as monaco from '@theia/monaco-editor-core';
+import { TaskContextService } from './task-context-service';
+
+export const SESSION_SUMMARY_VARIABLE: AIVariable = {
+    id: 'session-summary',
+    description: 'Resolves to a summary of the session with the given ID.',
+    name: 'session-summary',
+    label: 'Session Summary',
+    iconClasses: codiconArray('clippy'),
+    isContextVariable: true,
+    args: [{ name: 'session-id', description: 'The ID of the session to summarize.' }]
+};
+
+@injectable()
+export class SessionSumaryVariableContribution implements FrontendVariableContribution, AIVariableResolver {
```

Comment:
I would rename the variable to "TaskContext" (and all related things such as the label provider)

Reply:
specifically also the variable the user types "#taskContext"